stdlib/SparseArrays/src/sparsevector.jl

```diff
@@ -718,17 +718,9 @@ function findnz(x::SparseVector{Tv,Ti}) where {Tv,Ti}
 
     count = 1
     @inbounds for i = 1 : numnz
-        if nzval[i] != 0
-            I[count] = nzind[i]
-            V[count] = nzval[i]
-            count += 1
-        end
-    end
-
-    count -= 1
-    if numnz != count
-        deleteat!(I, (count+1):numnz)
-        deleteat!(V, (count+1):numnz)
+        I[count] = nzind[i]
+        V[count] = nzval[i]
+        count += 1
```

Comment:
No need for summing `count` anymore, either.

Reply:
Indeed. I copied the `SparseMatrix` method too litterally.